### PR TITLE
ENH: Support parsing Fortran abstract interface blocks.

### LINKED ISSUE #18403: Error when parsing fortran code with abstract interface
f2py appears to be missing support for abstract interfaces. Attempting to parse code containing an abstract interface produces an error.

### Reproducing code example:

```python
from numpy.f2py.crackfortran import crackfortran

crackfortran('test_interface.F90')
```

test_interface.F90:
```fortran
module test
  abstract interface
  end interface
end module test
```

### Error message:

```
Traceback (most recent call last):
  File "/Users/jhaiduce/Development/lightda/test_interface_crackfortran.py", line 3, in <module>
    crackfortran('test_interface.F90')
  File "/usr/local/lib/python3.9/site-packages/numpy/f2py/crackfortran.py", line 3310, in crackfortran
    readfortrancode(files, crackline)
  File "/usr/local/lib/python3.9/site-packages/numpy/f2py/crackfortran.py", line 526, in readfortrancode
    dowithline(finalline)
  File "/usr/local/lib/python3.9/site-packages/numpy/f2py/crackfortran.py", line 759, in crackline
    raise Exception('crackline: groupcounter(=%s) is nonpositive. '
Exception: crackline: groupcounter(=0) is nonpositive. Check the blocks.
```

### NumPy/Python version information:

1.20.1 3.9.1 (default, Feb  3 2021, 07:04:48) 
[Clang 11.0.0 (clang-1100.0.33.17)]

### PR DESCRIPTION
**Pull request by pearu**
Closes #18403

### CODE REVIEW

File: numpy/f2py/crackfortran.py [hunk 1]
```diff
@@ -1010,7 +1015,7 @@ def analyzeline(m, case, line):
         groupname[groupcounter] = block
         groupcache[groupcounter]['block'] = block
         if not name:
-            name = 'unknown_' + block
+            name = 'unknown_' + block.replace(' ', '_')
```
Comment (charris) [inline numpy/f2py/crackfortran.py:1018]: Can there be multiple spaces?

File: numpy/f2py/tests/test_abstract_interface.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          integer, intent(in) :: x, y
+          integer, intent(out) :: z
+          z = x + 2 * y
+        end subroutine
+        """)
+
+    def test_abstract_interface(self):
+        assert self.module.ops_module.foo(3, 5) == (8, 13)
+
+    def test_gh18403(self, tmp_path):
```
Comment (charris) [inline numpy/f2py/tests/test_abstract_interface.py:51]: Would be better to make the name more informative, e.g., `test_abstract_interface`.

Comment (charris) [reply numpy/f2py/tests/test_abstract_interface.py:51]: The issue can be put in a comment line.

**Comment by charris** [comment]
LGTM, just a naming nit.

File: numpy/f2py/tests/test_abstract_interface.py (hunk 2 above)
Comment (pearu) [reply numpy/f2py/tests/test_abstract_interface.py]: Fixed. Using test_parse_abstract_interface.

**Comment by pearu** [comment]
codecov/patch failure is not related to the changes in this PR.

**Pull request merged**